etc/shadow-maint/userdel-pre.d/01-kill_user_procs.sh

```diff
@@ -4,14 +4,17 @@ PATH="/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin"
 
 # Check user exists, and if so, send sigkill to processes that the user owns
 
-RUNNING=`ps -eo user | grep -Fx "$SUBJECT" | wc -l`
-
-# if the user does not exist, RUNNING will be 0
-
-if [ "${RUNNING}x" = "0x" ]; then
-  exit 0
+ps -eo user >/dev/null 2>&1
+if [ $? -eq 0 ]; then
```

Comment:
I looked at this a few days ago.  Thanks for pointing out the issue - we should do something, but I don't think that this is the best way to solve it.  I'd prefer there be one way to do it.

@edneville do you have any suggestions?

Reply:
I've tried to add a few pieces of code to adapt to the ps command with different options, but it doesn't cover all user scenarios and the code is redundant. So I think if ps -eo doesn't work, go straight to traversal. The userdel may take a little longer, but is acceptable as a low frequency operation.